core/src/main/java/org/apache/carbondata/core/datamap/DataMapChooser.java

```diff
@@ -269,6 +270,12 @@ private void extractColumnExpression(Expression expression,
       List<ColumnExpression> columnExpressions) {
     if (expression instanceof ColumnExpression) {
       columnExpressions.add((ColumnExpression) expression);
+    } else if (expression instanceof MatchExpression) {
+      // fake ColumnExpression for lucene. normal query string is alike "column:query term"
```

Comment:
please describe what is the use of this fake ColumnExpression.
And it is better to mention `Lucene query string` instead of `normal query string`